app/controllers/bank_accounts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BankAccountsController < ApplicationController
+  def index
+    if current_user.banker ==true || current_user.staff == true
+      @accounts = BankAccount.all
+      characters = Character.all
+      @characters_without_accounts = []
+      characters.each do |character|
+        if !BankAccount.has_account?(character.id) == true && character.archived==false
+          @characters_without_accounts << character
+        end
+      end
```

Comment:
It is *possible* to get this collection directly, with a single `ActiveRecord` query (hint: it uses the `.where.not()` syntax.) This would be a performance improvement over looping through all the characters in the system -- at minimum, I'd like you to refactor this to not use `Character.all`, given that you immediately discard archived characters. If you still end up using a `.each` loop, I won't hold it against you.

Reply:
could you provide a little more direction. I've tried these (and other vartiations)
Character.where(character.bank_account:nil)
Character.where.not(BankAccount.character_id:Character.id)

But I'm not sure how to write what you've suggested, and the docs aren't really clear to me.

Reply:
`@characters_without_account = Character.includes(:bank_account).where(archived: false, bank_accounts: { id: nil })` is probably the simplest implementation that works (it totally doesn't use .where.not()`, though, I'm apparently a liar.)